### PR TITLE
Alignment and summary

### PR DESCRIPTION
**Pull request by haydentwestbrook**
Creating a new pull request off a feature branch instead of off my master branch.

### CODE REVIEW

File: yax/artifacts/summary.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from yax.state.type.artifact import Artifact
+from io import StringIO
+import os
+import os.path
+import pdfkit
```
Comment (ebolyen) [inline yax/artifacts/summary.py:3]: Add this to `setup.py`

File: yax/artifacts/coverage_data.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from yax.state.type import Artifact
+from os import listdir
+from os.path import isfile, join, splitext
```
Comment (ebolyen) [inline yax/artifacts/coverage_data.py]: stylistically, these functions are not placed directly in the namespace. it's longer, but usually you see `os.path.join`

seeing `join` on it's own is unusual.

File: yax/artifacts/coverage_data.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                        tabs = line.split('\t')
+                        gi = tabs[2].split('|')[1]
+                        position = tabs[3]
+                        length = len(tabs[9])
+                        sample.alignments.append(self.Alignment(gi, length,
+                                                                position))
+                except Exception:
+                    continue
+
+    class Sample:
```
Comment (ebolyen) [inline yax/artifacts/coverage_data.py]: These are good, but a `namedtuple` will be dramatically more efficient. They are essentially structs.

File: yax/artifacts/summary.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        pdfkit.from_string(writer.getvalue(), str(self.data_dir) + "/" +
+                           str(file_name) + ".pdf")
+        writer.close()
+
+    def __complete__(self):
+        """
+        Clean up data_dir by deleting temporary coverage plot images
+        """
+        images = [self.data_dir + '/' + f for f in os.listdir(self.data_dir) if
```
Comment (ebolyen) [inline yax/artifacts/summary.py]: There will be a `working_dir` available to the module.

Comment (ebolyen) [inline yax/artifacts/summary.py]: which is automatically GCed by the state-system

File: yax/modules/alignment/alignment.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+def run_alignment(gi_references, sample_reads, output,
+                  bowtie_options, working_dir):
+    """
+    :param gi_references: Location of gi references
+    :param sample_reads: Location of reads trimmed by ReadPrep
+    :param output: Artifact to fill out with sam file alignments
+    :param bowtie_options: Optional bowtie parameters
+    """
+
+    command = ["bowtie2-build", gi_references, working_dir + "/index"]
```
Comment (ebolyen) [inline yax/modules/alignment/alignment.py]: `os.path.join(working_dir, "/index")`

File: yax/modules/alignment/alignment.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    try:
+        # Build bowtie indexes
+        subprocess.call(command)
+    except Exception as e:
+        print("ALIGNMENT: Failed to build bowtie indexes.")
```
Comment (ebolyen) [inline yax/modules/alignment/alignment.py]: should this situation be tracked?

Comment (MikeDeberg) [inline yax/modules/alignment/alignment.py]: Actually I'm building bowtie2 indexes as well, maybe this should be a util?

**Pull request merged**